polling_stations/apps/addressbase/helpers.py

```diff
@@ -18,6 +19,84 @@ def centre_from_points_qs(qs):
     return poly.centroid
 
 
+AddressTuple = namedtuple('Address', [
+    'address',
+    'postcode',
+    'council',
+    'polling_station_id',
+    'slug'])
+
+
+class AddressSet:
+
+    def __len__(self):
+        return len(self.elements)
+
+    def __init__(self):
+        self.elements = set()
+
+    def add(self, address):
+        self.elements.add(address)
+
+    def save(self, batch_size):
+
+        addresses_db = []
+        for address in self.elements:
+            record = ResidentialAddress(
+                address=address.address,
+                postcode=address.postcode,
+                polling_station_id=address.polling_station_id,
+                council=address.council,
+                slug=address.slug,
+            )
+            addresses_db.append(record)
+
+        ResidentialAddress.objects.bulk_create(
+            addresses_db, batch_size=batch_size)
+
+
+class EdgeCaseFixer:
+
+    def __init__(self, target_council_id):
+        self.address_set = AddressSet()
+        self.target_council_id = target_council_id
+
+    def make_addresses_for_postcode(self, postcode):
+        addresses = Address.objects.filter(postcode=postcode)
+        for address in addresses:
+            try:
+                district = PollingDistrict.objects.get(
+                    area__covers=address.location, council_id=self.target_council_id)
+            except PollingDistrict.DoesNotExist:
+                # Chances are this is on the edge of the council area, and
+                # we don't have data for the are the property is in
+                continue
+            except PollingDistrict.MultipleObjectsReturned:
```

Comment:
Is it worth logging overlapping districts? It could point to fixable problems, like clipping district boundaries by council boundaries. (Also typo on the line below: s/the/that/)